src/Sentry/Internal/Hub.cs

```diff
@@ -182,11 +186,38 @@ public void StartSession()
             }
         }
 
-        public void EndSession(SessionEndStatus status = SessionEndStatus.Exited)
+        public void PauseSession()
+        {
+            lock (_sessionPauseLock)
+            {
+                _sessionPauseTimestamp = DateTimeOffset.Now;
+            }
+        }
+
+        public void ResumeSession()
+        {
+            lock (_sessionPauseLock)
+            {
+                var pauseDuration = (DateTimeOffset.Now - _sessionPauseTimestamp).Duration();
```

Comment:
Does it make sense to provide ISystemClock to the hub for testability, to fake the pause?